modules/python/python-types.c

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright (c) 2022 One Identity
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "python-types.h"
+#include "python-helpers.h"
+#include "scanner/list-scanner/list-scanner.h"
+#include "scratch-buffers.h"
+#include "str-repr/encode.h"
+#include "messages.h"
+
+static void
+_log_py_object_construction_error(void)
+{
+  gchar buf[256];
+  _py_format_exception_text(buf, sizeof(buf));
+  msg_error("Error while constructing python object", evt_tag_str("exception", buf));
+  _py_finish_exception_handling();
+}
+
+static void
+_log_py_object_conversion_error_with_exception(void)
+{
+  gchar buf[256];
+  _py_format_exception_text(buf, sizeof(buf));
+  msg_error("Error while converting python object", evt_tag_str("exception", buf));
+  _py_finish_exception_handling();
+}
+
+static void
+_log_py_object_conversion_error_with_invalid_type(PyObject *obj, const gchar *expected_type)
+{
+  msg_error("Error while converting python object",
+            evt_tag_str("expected_type", expected_type),
+            evt_tag_str("actual_type", Py_TYPE(obj)->tp_name));
+}
+
+PyObject *
+py_bytes_from_string(const char *value, gssize len)
+{
+  PyObject *obj = NULL;
+
+  if (len < 0)
```

Comment:
I would simply not allow to call this without proper len, as far as I can see all callee currently provides len as well.
If the value is provided without len, it must be zero terminated and the usual `if (len==-1) len = strlen(value);` could be used.

Reply:
Fixed this, and another similar case.